zebra/rt_netlink.c

```diff
@@ -246,6 +246,34 @@ static vrf_id_t vrf_lookup_by_table(uint32_t table_id, ns_id_t ns_id)
 	return VRF_DEFAULT;
 }
 
+/**
+ * @parse_encap_mpls() - Parses encapsulated mpls attributes
+ * @tb:         Pointer to rtattr to look for nested items in.
+ * @labels:     Pointer to store labels in.
+ *
+ * Return:      Number of mpls labels found.
+ */
+static int parse_encap_mpls(struct rtattr *tb, mpls_label_t *labels)
+{
+	struct rtattr *tb_encap[MPLS_IPTUNNEL_MAX + 1];
+	mpls_lse_t *lses = NULL;
+	int num_labels = 0;
+	uint32_t ttl = 0;
+	uint32_t bos = 0;
+	uint32_t exp = 0;
+	mpls_label_t label = 0;
+
+	memset(tb_encap, 0, sizeof(struct rtattr *) * MPLS_IPTUNNEL_MAX + 1);
```

Comment:
`MPLS_IPTUNNEL_MAX + 1` needs to be inside parentheses to avoid an operator precedence error.

My comment about using `memset` was outdated and Stephen had already addressed the issue by initializing `tb_encap` by using `tb_encap = {0}`, which is my opinion is better than using `memset` in terms of code readability.

Other than that this PR looks good to me!